src/mmtk.rs

```diff
@@ -142,7 +142,10 @@ impl<VM: VMBinding> MMTK<VM> {
     }
 
     pub fn harness_begin(&self, tls: VMMutatorThread) {
-        // FIXME Do a full heap GC if we have generational GC
+        // For generational plans, force a full heap collection.
+        if let Some(gen) = self.plan.generational() {
+            gen.force_full_heap_collection();
```

Comment:
What about passing another argument to the `handle_user_collection_request` call below to force a full GC?

Reply:
That sounds good. But we can only add the argument in our internal call, rather than expose the idea of full heap GC to the user.

Reply:
Done